core/platform/storage/cloud_storage_services.py

```diff
@@ -0,0 +1,125 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Provides translate_text functionality from Google Cloud Translate."""
+
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+import functools
+
+from google.cloud import storage
+
+
+@functools.lru_cache(maxsize=1)
+def get_client():
+    """Gets Cloud Storage client.
+
+    Returns:
+        storage.Client. Cloud Storage client.
+    """
+    return storage.Client()
+
+
+def isfile(bucket_name, filepath):
+    """Checks if the file with the given filepath exists in the GCS.
+
+    Args:
+        bucket_name: str. The name of the GCS bucket.
+        filepath: str. The path to the relevant file within the entity's
+            assets folder.
+
+    Returns:
+        bool. Whether the file exists in GCS.
+    """
+    return get_client().get_bucket(bucket_name).get_blob(filepath) is not None
+
+
+def get(bucket_name, filepath):
+    """Gets a file as an unencoded stream of raw bytes.
+
+    Args:
+        bucket_name: str. The name of the GCS bucket.
+        filepath: str. The path to the relevant file within the entity's
+            assets folder.
+
+    Returns:
+        FileStream or None. It returns FileStream domain object if the file
+        exists. Otherwise, it returns None.
+    """
+    blob = get_client().get_bucket(bucket_name).get_blob(filepath)
+    data = blob.download_as_bytes()
+    return data
+
+
+def commit(bucket_name, filepath, raw_bytes, mimetype):
+    """Commit raw_bytes to the relevant file in the entity's assets folder.
+
+    Args:
+        bucket_name: str. The name of the GCS bucket.
+        filepath: str. The path to the relevant file within the entity's
+            assets folder.
+        raw_bytes: str. The content to be stored in the file.
+        mimetype: str. The content-type of the cloud file.
+    """
+    blob = get_client().get_bucket(bucket_name).blob(filepath)
+    blob.upload_from_string(raw_bytes, content_type=mimetype)
+
+
+def delete(bucket_name, filepath):
+    """Deletes a file and the metadata associated with it.
+
+    Args:
+        bucket_name: str. The name of the GCS bucket.
+        filepath: str. The path to the relevant file within the entity's
+            assets folder.
+    """
+    blob = get_client().get_bucket(bucket_name).get_blob(filepath)
+    blob.delete()
+
+
+def copy(bucket_name, source_assets_path, dest_assets_path):
+    """Copy images from source_path.
+
+    Args:
+        bucket_name: str. The name of the GCS bucket.
+        source_assets_path: str. The path to the source entity's assets
+            folder.
+        dest_assets_path: str. The path to the relevant file within the entity's
+            assets folder.
+    """
+    src_blob = get_client().get_bucket(bucket_name).get_blob(source_assets_path)
```

Comment:
You do the computation `get_client().get_bucket(bucket_name)` many times, maybe extract that logic out into a variable.

Even better, perhaps do that across all the functions and add a private module-level _get_bucket() method.

Reply:
Done.